src/sage/rings/polynomial/integer_valued_polynomials.py

```diff
@@ -0,0 +1,1125 @@
+                """
+                return self.algebra_generators()[0]
+
+            @cached_method
+            def algebra_generators(self):
+                r"""
+                Return the generators of this algebra.
+
+                EXAMPLES::
+
+                    sage: A = IntegerValuedPolynomialRing(ZZ).S(); A
+                    Integer-Valued Polynomial Ring over Integer Ring
+                    in the shifted basis
+                    sage: A.algebra_generators()
+                    Family (S[1],)
+                """
+                NonNeg = self.basis().keys()
+                return Family([self.monomial(NonNeg(1))])
+
+            gens = algebra_generators
+
+        class ElementMethods:
+            def __call__(self, v):
+                """
+                Evaluation at some value ``v``
+
+                EXAMPLES::
+
+                     sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                     sage: B = F.gen()
+                     sage: f = B**2+4*B+6
+                     sage: f(1/3)
+                     118/9
+
+                     sage: F = IntegerValuedPolynomialRing(ZZ).B()
+                     sage: B = F.gen()
+                     sage: f = B**2+4*B+6
+                     sage: f(1/3)
+                     67/9
+                """
+                return self.polynomial()(v)
+
+            def shift(self, j=1):
+                """
+                Shift all indices by `j`.
+
+                INPUT:
+
+                - `j` -- integer (default: 1)
+
+                In the binomial basis, the shift by 1 corresponds to
+                a summation operator from `0` to `x`.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).B()
+                    sage: B = F.gen()
+                    sage: (B+1).shift()
+                    B[1] + B[2]
+                    sage: (B+1).shift(3)
+                    B[3] + B[4]
+                """
+                A = self.parent()
+                return A._from_dict({i + j: c for i, c in self})
+
+    class Shifted(CombinatorialFreeModule, BindableClass):
+        r"""
+        The integer-valued polynomial ring in the shifted basis.
+
+        Integer-valued polynomial rings are commutative and associative
+        algebras, with a basis indexed by non-negative integers.
+
+        The basis used here is given by `B[i] = \binom{i+n}{i}` for `i \in \NN`.
+
+        Assuming `n_1 \leq n_2`, the product of two monomials `B[n_1] \cdot B[n_2]`
+        is given by the sum
+
+        .. MATH::
+
+            \sum_{k=0}^{n_1} (-1)^k \binom{n_1}{k}\binom{n_1+n_2-k}{n_1} B[n_1 + n_2 - k].
+
+        There is a conversion formula between the two bases
+
+        .. MATH::
+
+            \binom{x}{i} = \sum_{k=0}^{i} (-1)^{i-k} \binom{i}{k} \binom{x+k}{k}.
+
+
+        REFERENCES:
+
+        - :wikipedia:`Integer-valued polynomial`
+
+        INPUT:
+
+        - ``A`` -- a parent with realization
```

Comment:
I find this useless to confusing for a user since they will never directly construct this by calling the class on its own. So I would remove it.